plugins/imjournal/imjournal.c

```diff
@@ -443,6 +443,8 @@ readjournal(void)
 		if (!sd_journal_get_cursor(j, &c)) {
 			free(last_cursor);
 			last_cursor = c;
+		} else {
+			free(c);
```

Comment:
I don't understand.  `c` is set to `NULL` at the beginning of the function.  It is not set anywhere else except by `sd_journal_get_cursor`. If you look at the implementation of that function https://github.com/systemd/systemd/blob/master/src/journal/sd-journal.c#L909 you can see that if `sd_journal_get_cursor` returns a non-zero value, then `c` is not set.  So there should be nothing to free.

Reply:
@richm 
You're right. Actually, I'm closing out this PR because I realize that the problem is downstream-only.